### PR TITLE
Stats rewrite yet again

### PR DESCRIPTION
**Pull request by gbasood**
(psst the web server part of this is here: https://github.com/gbasood/VGStats)

Finally getting around to some of this. Don't get your hopes up about the site being updated, I'm just putting in work as motivation happens right now.

This is so far just a bunch of cleanup. Next up is having roles and factions report their own data, which will get added to the rest of the stats collection and serialized for use in the stats server.

I tried looking for statistics-related issues but couldn't find any. If you have suggestions **specifically related to data that is not yet collected or needs to be collected differently (bugs, non-useful data, etc)** you could suggest them here in this PR or over at https://github.com/gbasood/vgstation-statistics-viewer

If you have suggestions about displaying data, https://github.com/gbasood/vgstation-statistics-viewer

## The current state of stats
- Duplicated data (pop snapshots being collected two different ways), which may have been something I told someone else to not fix before, whoops
- Collecting antag data is a pain and requires lots of tedious spaghetti code

## After this PR
- ~~Duplicated data removed~~ done
- Antag-specific data gathered in a decentralized way that allows easier revision
- **NEW** - Game code defines whether a role or faction is considered as having 'won', minor or otherwise. Uses pre-existing logic to do so. This gives you guys more control away from the web server. 
  - Currently the logic is limited to "win all your objectives or you fail." I think that's fair, going forward. Players will be able to see individual roles/objectives completion status on the web site as well.
- **NEW** - End-of-round manifest export. Contains player key, name, and assignment.

Role and faction checklist:
- [x] Traitor
- [x] Malf
- [x] Vamp
  - [x] Power purchases
  - [x] Blood total
- [x] Blob
 - Now includes **structure counts**, split into 3 categories:
   - Unique sums of each structure type built over the course of the entire round (i.e. counts separated by structure type)
   - Sum of blob structures at the end of the round
   - Sum of all blob structures of any kind at the end of the round.
- [x] Wizard
 - Now includes **all spellbook purchases**
- [x] Revs
 - Logs recruits converted for both revs and revheads, and logs flashes created for revheads.
- [x] ~~Nukeops~~ this is fine, all relevant data is already gathered
- [x] Ninja
 - I'll be honest I still haven't seen a ninja in play, so I just grabbed some things that looked interesting.
 - Num of shuriken thrown, num of times charged sword, num of times stealth posters posted
- [x] Catbeast
 - Ticks survived, threat generated, threat level inflated and areas defiled are all good values.
- **Legacy cult-specific stats no longer tracked.**

### CODE REVIEW

File: code/datums/gamemode/role/ninja.dm
Comment (Kurfursten) [inline code/datums/gamemode/role/ninja.dm]: this applies to everything, but why increment it on the role with a new var instead of just going on through to get the stat datum and directly incrementing it there?

Comment (gbasood) [reply code/datums/gamemode/role/ninja.dm]: Honestly, I can't remember. I think I was trying to be minimally invasive to faction/role datums logic itself, but I've accomplished the opposite here, in the end. I'll probably change it back unless I can remember why I did this.

Comment (gbasood) [reply code/datums/gamemode/role/ninja.dm]: Reviewed my code. Already posted in Discord but I'll put it here as well. Basically I did this so that within faction/role logic, you wouldn't have to type check a stat datum to access its variables every time. The statistics are instead stored on the role/faction datum itself, then on `New()` the faction/role stat specific datum just grabs the data already on the game's role/faction. No type checks involved.

Is it worth keeping? Is one better than the other? I have no clue!

**Comment by gbasood** [comment]
Still untested.

**Review by Kurfursten — APPROVED** [review]
Looks good to me

**Comment by gbasood** [comment]
Fixed conflict.
@d3athrow Once this gets merged you can delete all existing stat files. I won't be processing anything before this PR.

EDIT: don't merge yet I need to fix that blob apparently never got updated to the new system

**Comment by gbasood** [comment]
ACTUALLY good to merge now.
For anyone curious, the web server side of this lives at https://github.com/gbasood/VGStats

**Pull request merged**
